src/azure/auto-install-azure-cli-module-decorator.ts

```diff
@@ -85,9 +85,8 @@ export class AutoInstallAzureCliModuleDecorator implements AzureCliFacade {
   }
 
   private isAzureModuleMissingError(e: Error): boolean {
-    return (e instanceof MeshAzurePlatformError) &&
-      (e as MeshAzurePlatformError).errorCode ===
-        ErrorCodes.AZURE_CLI_MISSING_EXTENSION;
+    return e instanceof MeshAzurePlatformError &&
+      e.errorCode === "AZURE_CLI_MISSING_EXTENSION";
```

Comment:
I was unsure how well string enums perform when transpiled down to JS. But my solution is also "typesafe" because it a compound string type. But I can replace it with an String enum. Seems like they are working fine.